packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -170,7 +171,40 @@ export function handleRedirects(
   event: InternalEvent,
   redirects: RedirectDefinition[],
 ): InternalResult | undefined {
-  const { internalEvent, __rewrite } = handleRewrites(event, redirects);
+  if (
```

Comment:
The return types here are mostly the same except for the `Location`, perhaps we can simplify the code later.